src/neo/Oracle/OraclePolicyContract.cs

```diff
@@ -0,0 +1,230 @@
+        private StackItem DelegateOracleValidator(ApplicationEngine engine, Array args)
+        {
+            ECPoint consignorPubKey = args[0].GetSpan().AsSerializable<ECPoint>();
+            ECPoint consigneePubKey = args[1].GetSpan().AsSerializable<ECPoint>();
+            UInt160 account = Contract.CreateSignatureRedeemScript(consignorPubKey).ToScriptHash();
+            if (!InteropService.Runtime.CheckWitnessInternal(engine, account)) return false;
+            StoreView snapshot = engine.Snapshot;
+            StorageKey key = CreateStorageKey(Prefix_Validator, consignorPubKey);
+            if (snapshot.Storages.TryGet(key) != null)
+            {
+                StorageItem value = snapshot.Storages.GetAndChange(key);
+                value.Value = consigneePubKey.ToArray();
+            }
+            else
+            {
+                snapshot.Storages.Add(key, new StorageItem
+                {
+                    Value = consigneePubKey.ToArray()
+                });
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Get current authorized Oracle validator.
+        /// </summary>
+        /// <param name="engine">VM</param>
+        /// <param name="args">Parameter Array</param>
+        /// <returns>Authorized Oracle validator</returns>
+        [ContractMethod(0_03000000, ContractParameterType.Array)]
+        private StackItem GetOracleValidators(ApplicationEngine engine, Array args)
+        {
+            return new Array(engine.ReferenceCounter, GetOracleValidators(engine.Snapshot).Select(p => (StackItem)p.ToArray()));
+        }
+
+        /// <summary>
+        /// Get current authorized Oracle validator
+        /// </summary>
+        /// <param name="snapshot">snapshot</param>
+        /// <returns>Authorized Oracle validator</returns>
+        public ECPoint[] GetOracleValidators(StoreView snapshot)
+        {
+            ECPoint[] oraclePubKeys = PolicyContract.NEO.GetValidators(snapshot);
+            for (int index = 0; index < oraclePubKeys.Length; index++)
+            {
+                var oraclePubKey = oraclePubKeys[index];
+                StorageKey key = CreateStorageKey(Prefix_Validator, oraclePubKey);
+                ECPoint delegatePubKey = snapshot.Storages.TryGet(key)?.Value.AsSerializable<ECPoint>(1);
+                if (delegatePubKey != null) { oraclePubKeys[index] = delegatePubKey; }
+            }
+            return oraclePubKeys;
+        }
+
+        /// <summary>
+        /// Get number of current authorized Oracle validator
+        /// </summary>
+        /// <param name="engine">VM</param>
+        /// <param name="args">Parameter Array</param>
+        /// <returns>The number of authorized Oracle validator</returns>
+        [ContractMethod(0_03000000, ContractParameterType.Integer)]
```

Comment:
How have you estimated these costs?

Reply:
Just a wayward value.hahhaha
We can discuss

Reply:
I have changed the value of the costs.If it is not reasonable, you can mention it again.